lib/lita/handlers/artifactory.rb

```diff
@@ -92,6 +114,35 @@ def promote(response)
         response.reply reply_msg
       end
 
+      def push(response)
+        ruby_gem      = response.args[2]
+        version       = response.args[3]
+        human_name    = "#{ruby_gem} gem version #{version}"
+        gem_source    = "#{config.endpoint}/api/gems/gems-local/"
+        missing_gem   = false
+
+        Dir.mktmpdir do |dir|
+          Dir.chdir(dir) do
+            %w{ruby universal-mingw32}.each do |platform|
+              unless system(fetch_command_for_platform(platform, ruby_gem, version, gem_source))
+                response.reply("There were errors retrieving the #{human_name} from #{gem_source}!")
+                missing_gem = true
+              end
+            end
+
+            break if missing_gem
+
+            Dir.glob("*.gem") do |gem_file|
+              if system("gem push #{gem_file} --key chef_rubygems_api_key")
```

Comment:
Is it worth using `mixlib-shellout` here and capturing the STDOUT/STDERR to return to the user on failures? It might be easier to use the Ruby wrapper for the RubyGems.org API:
https://github.com/rubygems/gems

```ruby
Gems.push(File.new('somegem-0.2.1.gem'))
```

That would allow you to easily catch `Net::HTTPServerException` errors (you'll def want to ensure that is the exact exception) and grab the error message for replying to the user.